src/history/hs_rec.c

```diff
@@ -213,10 +219,24 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         error_on_ts_ordering = false;
     }
 
-    if (ret == 0)
-        WT_ERR(__hs_delete_reinsert_from_pos(session, cursor, btree->id, key, tw->start_ts + 1,
-          true, false, error_on_ts_ordering, &counter, tw));
-
+    if (ret == 0) {
+        WT_ASSERT(session, tw->start_ts + 1 > WT_TS_NONE);
+        /* Get the history store cursor position. */
+        WT_ERR_NOTFOUND_OK(__hs_cursor_pos(session, cursor, btree->id, key, tw->start_ts + 1,
+                             error_on_ts_ordering, tw, &non_ts_updates),
+          true);
+
```

Comment:
Can you add a bigger comment here to explain the different cases?

Reply:
Hi Chenhao, the case for inserts is simple, skip reinsert for non-timestamped updates. I added an extra line, there is a more detailed comment for the delete pathway. Let me know what else you would like to see. [b6900cf](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/b6900cf433be480879bbfe2be8d51f4507fcea13)